app/controllers/items_controller.rb

```diff
@@ -0,0 +1,48 @@
+class ItemsController < ApplicationController
+  def index
+    if params[:merchant_id] == nil
+      @items = Item.all
+    else
+      @merchant = Merchant.find(params[:merchant_id])
+      @items = @merchant.items
+    end
+  end
+
+  def show
+    @item = Item.find(params[:id])
+  end
+
+  def new
+    @merchant_id = params[:merchant_id]
+  end
+
+  def create
+    merchant = Merchant.find(params[:merchant_id])
+    item = Item.create!(item_params)
+    item.save!
+    redirect_to "/items/#{item.id}"
+  end
+
+  def edit
+    @item = Item.find(params[:id])
+  end
+
+  def update
+    @item = Item.find(params[:id])
+      if @item.update_attributes(item_params)
```

Comment:
Here you started to do a sad path, but didn't quite finish it. You need an `else` in here to deal with when the update fails. A good thing to do in that case would be to show the user the form again and show a flash message describing what went wrong.